### DISCUSSION TITLE
Is it possible to video output through the Raspberry Pi's VideoCore?

### DISCUSSION THREAD

**Discussion by Mimizusan3373** [zephyrproject-rtos/zephyr#79157] [Q&A]
thank you for the amazing RTOS!

Is it possible to use Zephyr to video output through the Raspberry Pi's VideoCore? I tried building the `./sample/display` for the `rpi_4b` board, but it failed.

**Comment by ghost** [comment]
This would allow many cool projects, but I see no mention of this anywhere. What would it take to implement?

E: this might be helpful, https://github.com/bztsrc/raspi3-tutorial/tree/master/09_framebuffer

https://github.com/Random06457/rpi4-gpu-bare-metal-examples

**Comment by moc67331** [comment]
I’ve implemented a display device driver for the BCM2711 (Raspberry Pi 4B) and managed to render some colors. However, the output is clearly incorrect. 😅
<img width="1481" alt="スクリーンショット 2025-01-12 13 48 00" src="https://github.com/user-attachments/assets/fa9df8b4-4393-4130-a061-ed012cfa3893" />

Is there anyone interested in working on this implementation? I plan to continue working on it in my free time, but it may take a while because I’m still not very familiar with Zephyr. (Maybe it just requires the [https://github.com/bztsrc/raspi3-tutorial/tree/master/09_framebuffer] that gergovari in the previous reply and adding DTS configuration and adding an MMU configuration via device_map().)

**Comment by moc67331** [comment]
a small LVGL demo worked in qemu for raspberry pi 4b, but it’s not perfect.

https://github.com/user-attachments/assets/5ceb2d8a-5973-46cb-b61b-8c39c8b9cd53
